### ISSUE TITLE
[ Feature request ] - Import existing files in a torrent download

### ISSUE THREAD

**Issue by gabrielklein** [transmission/transmission#4562]
### What is the issue?

Hello, Thank you for the great work!

It would be great to be able to import existing local files in an existing download.

I often download updated version of existing torrent.

Example, in 2022 I've downloaded Stuff.2022.torrent
Torrent Stuff.2022: File a.2022, file b.2022, file c.2022, d.2021

Then I download in 2023 Stuff.2023.torrent
Torrent Stuff2023: File a.2022, file b.2022, file c.2022.with.modified.content, e.2023

To avoid downloading the same stuff again and again, I start the download of "Stuff.2023.torrent", then pause it, manually copy my old stuff (stuff.2022) on my stuff.2023 folder, click "verify local data", then start again the torrent.

But It would be great if just under "verify local data", I have the option to "import existing files" that allow me to select a folder on my computer and start updating my current download with files that are already in this folder.

### Which application of Transmission?

Qt app on Linux, BSD, etc.

### Which version of Transmission?

3.0.0

**Comment by ckerr** [comment]
Thanks for the suggestion!

This is one of those feature requests that I'm never sure how to handle --

- On the one hand, a good idea. I'd probably use it myself.

- On the other hand, this is probably something very few people would use except for some power users. Since there's the existing workaround you described. And someone who hits this use case frequently could write a script to add torrents like this, e.g. add + pause + copy files + verify + start.

So even though I'm not going to put time into this, I'm not against the idea. If someone else wants to put the time & effort into this I'd be happy to do code review.

**Comment by killemov** [comment]
I agree with NOT handling this with Transmission, especially since the easiest solution is to just add a symbolic link from Stuff.2023 to Stuff.2022. (OR ...  rename the path part of the torrent in TM. OR ... rename the old folder on disk and remove Stuff.2022 from TM. But symbolic linking is safer.)
